### PR TITLE
New Element Individuals to handle NONE and NOASSERTION scenarios for the 'to' property of Relationships

### PR DESCRIPTION
**Pull request by sbarnum**
Addresses #527

### CODE REVIEW

**Review by kestewart — APPROVED** [review]
Thanks for pulling this together Sean.  Looks good to me.

**Review by goneall — APPROVED** [review]
This looks good to me - The description of the NoAssertion captured all the cases I was looking for.

Thanks @sbarnum

File: model/Core/Classes/Relationship.md [hunk 1]
```diff
@@ -11,6 +11,12 @@ Describes a relationship between one or more elements.
 A Relationship is a grouping of characteristics unique to an assertion
 that one Element is related to one or more other Elements in some way.
 
+To explicitly assert that no such relationships exist, the `to` property 
+should contain the 'NONE' individual.
```
Comment (maxhbr) [inline model/Core/Classes/Relationship.md:15]:
```suggestion
should contain the 'NONE' individual and no other elements.
A relationship that contains 'NONE' and other elements in the `to` property is not valid.
```

File: model/Core/Classes/Relationship.md [hunk 2]
```diff
@@ -11,6 +11,12 @@ Describes a relationship between one or more elements.
...
+
+To explicitly assert that no assertions are being made regarding the 
+existence of such relationships, the `to` property should contain the 'NOASSERTION' individual.
```
Comment (maxhbr) [inline model/Core/Classes/Relationship.md:18]:
```suggestion
existence of such relationships, the `to` property should also contain the 'NOASSERTION' individual.
```

**Review by maxhbr — CHANGES_REQUESTED** [review]
Added some more explanation, especially that `[Element1, NONE]` is not allowed. We should make that even more explicit so that validation also catches that.

There is also the case of a Relationship with `from`=NOASSERTION, `relationshipType`="ancestorOf", and `to`=Element1. This should also be explained or should be disallowed. Same for NONE.

File: model/Core/Classes/Relationship.md (hunk 1 above)
Comment (sbarnum) [reply model/Core/Classes/Relationship.md:15]: This looks like a good added clarification.

File: model/Core/Classes/Relationship.md (hunk 2 above)
Comment (sbarnum) [reply model/Core/Classes/Relationship.md:19]: This change does not look appropriate to me. Maybe I am misunderstanding its intent?
If the `to` property contains a value of NOASSERTION it should contain no other values similar to NONE.
If it does contain contain ANY other values then assertions are being made and NOASSERTION would not hold true.

I believe the clarification made for NONE ("and not other elements")should also be made here.

Comment (kestewart) [reply model/Core/Classes/Relationship.md:19]: Discssed with Max on call, and we'll not apply this suggestion from Max.

We'll go ahead and merge this as is now.

**Comment by maxhbr** [comment]
I disagree that this is the right direction. I think a to relationship towards an empty list, together with relationship completeness, is a better way to express that information.

**Comment by kestewart** [comment]
Discussion is we should document from migration guide, how to go this for 3.0.   Concern is this with completeness is going to cause more confusion than it resolves.   Let's document it, and then decide which PR to apply.   ( Documenting Empty list is ok with completeness indicator..  vs.  adding this in again)

**Comment by goneall** [comment]
Decided to move this to post RC2 and go with the existing solution using the `completness` property for RC2.

**Comment by kestewart** [comment]
Discussed 623 with @JPEWdev and agreed we're going to merge it.   We're going to raise this in next tech call.

**Comment by goneall** [comment]
From tech call on 2 April 2024: agreed to rename `None` and `NoAssertion` to `NoneElement` and `NoAssertionElement` resp.

**Review by goneall — CHANGES_REQUESTED** [review]

**Review by goneall — CHANGES_REQUESTED** [review]
Per tech call on 2 April 2024 -

**Review by goneall — APPROVED** [review]

**Review by zvr — APPROVED** [review]

**Pull request merged**